bundles/org.openhab.core/src/test/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunctionTest.java

```diff
@@ -380,4 +385,160 @@ private GroupItem createGroupItem(String name, Class<? extends Quantity<?>> dime
         item.setState(state);
         return item;
     }
+
+    @ParameterizedTest
+    @MethodSource("locales")
+    public void testSumFunctionQuantityTypeDifferentUnitsWithBaseItem(Locale locale) {
+        Locale.setDefault(locale);
+
+        Set<Item> items = new LinkedHashSet<>();
+        items.add(createNumberItem("TestItem1", Temperature.class, new QuantityType<>("23.54 °C")));
+        items.add(createNumberItem("TestItem2", Temperature.class, UnDefType.NULL));
+        items.add(createNumberItem("TestItem3", Temperature.class, new QuantityType<>("192.2 °F")));
+        items.add(createNumberItem("TestItem4", Temperature.class, UnDefType.UNDEF));
+        items.add(createNumberItem("TestItem5", Temperature.class, new QuantityType<>("395.56 K")));
+
+        Item baseItem = new NumberItem("Number:Temperature", "BaseItem", kelvinProvider);
+        GroupFunction function = new QuantityTypeArithmeticGroupFunction.Sum(Temperature.class, baseItem);
+        State state = function.calculate(items);
+
+        assertEquals(new QuantityType<>("1054.40 K"), state);
+    }
+
+    @ParameterizedTest
+    @MethodSource("locales")
+    public void testAvgFunctionQuantityTypeDifferentUnitsWithBaseItem(Locale locale) {
+        Locale.setDefault(locale);
+
+        Set<Item> items = new LinkedHashSet<>();
+        items.add(createNumberItem("TestItem1", Temperature.class, new QuantityType<>("100 °C")));
+        items.add(createNumberItem("TestItem2", Temperature.class, UnDefType.NULL));
+        items.add(createNumberItem("TestItem3", Temperature.class, new QuantityType<>("113 °F")));
+        items.add(createNumberItem("TestItem4", Temperature.class, UnDefType.UNDEF));
+        items.add(createNumberItem("TestItem5", Temperature.class, new QuantityType<>("294.15 K")));
+
+        Item baseItem = new NumberItem("Number:Temperature", "BaseItem", kelvinProvider);
+        GroupFunction function = new QuantityTypeArithmeticGroupFunction.Avg(Temperature.class, baseItem);
+        State state = function.calculate(items);
+
+        assertEquals(new QuantityType<>("55.33333333333333333333333333333334 °C"), state);
```

Comment:
This is prone to machine rounding differences when running the test. Try converting the state to °C, take the double value and use the assertEquals method with an extra argument (the tolerance).

Reply:
Done